src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -504,6 +506,22 @@ export class ConnectionWidget extends lifecycle.Disposable {
 			this._tableContainer.classList.remove('hide-username');
 			this._tableContainer.classList.add('hide-password');
 			this._tableContainer.classList.remove('hide-azure-accounts');
+		} else if (currentAuthType === AuthenticationType.dSTSAuth) {
+			this._accountManagementService.getAccountsForProvider('dstsAuth').then(accounts => {
+				if (accounts && accounts.length > 0) {
+					accounts[0].key.providerArgs = {
+						serverName: this.serverName,
+						databaseName: this.databaseName
+					};
+
+					this._accountManagementService.getAccountSecurityToken(accounts[0], '', null).then(securityToken => {
+						this._token = securityToken.token;
```

Comment:
Hmm, so this is all done async'ly but the token getter below assumes it's been set. What happens if the user clicks before this has a chance to finish populating? You should make sure that we handle that nicely.